### PR TITLE
core: fix register test

### LINKED ISSUE #467: core: several tests in TestRegister time out instead of indended error
https://github.com/decred/dcrdex/blob/e572dd5ff75f09c1e294d0fe77472ba1e50d1f4e/client/core/core_test.go#L1155-L1185

Some of these test timeout after request timeout, but I think they should be returning errors relevant to the test conditions.  It seeems some responses need to be queued up, but I'm not sure.

### PR DESCRIPTION
**Pull request by buck54321**
Resolves #467

Fixes bugs and adds coverage. Adds coded errors that can be used for ID.

### CODE REVIEW

File: client/core/core_test.go [hunk 1]
```diff
@@ -1361,7 +1431,7 @@ func TestConnectDEX(t *testing.T) {
 	}
 
 	// Bad URL.
-	ai.Host = ":::"
+	ai.Host = string([]byte{0x7f}) // Illegal ASCIII control character
```
Comment (buck54321) [inline client/core/core_test.go:1434]: I'm not certain what error path this was supposed to hit, but it wasn't.

**Comment by chappjc** [comment]
All of client/core tests in 1.25s now, nice.  Sort of related to the register test are `(*Core).GetFee`, which has no test coverage right now.  Would you want to role that coverage into this PR?  I think it's just a success path, an already-registered/`dupeDEXErr` error, and a connection/`connectionErr` test.

File: client/core/errors.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		s:    err.Error(),
+		code: code,
+	}
+}
+
+// errorHasCode checks whether the error is an Error and has the specified code.
+func errorHasCode(err error, code int) bool {
+	var e *Error
+	return errors.As(err, &e) && e.code == code
+}
```
Comment (JoeGruffins) [inline client/core/errors.go:58]: What do you think about the proposed `error` best practices in the dcrd repo? https://github.com/decred/dcrd/issues/2181

**Comment by JoeGruffins** [comment]
What do you think about the proposed `error` best practices in the dcrd repo? https://github.com/decred/dcrd/issues/2181

**Comment by chappjc** [comment]
> What do you think about the proposed `error` best practices in the dcrd repo? [decred/dcrd#2181](https://github.com/decred/dcrd/issues/2181)

I do think we should have a dex error type that works similarly.  Perhaps the following error type used in the db/drivers/pg package could be the basis for such a type, `dex.DexError` perhaps: https://github.com/decred/dcrdex/blob/ecbeacc290e78a1b0a199260c2effc0acd9dc0d6/server/db/driver/pg/errors.go#L14-L38

**Comment by chappjc** [comment]
But to clarify, I don't think we need to make a common dex error type like this in this PR.

**Comment by buck54321** [comment]
Added `TestGetFee` and cleaned up a little bit. I think these coded errors can probably be used in all core tests. Will follow up.

**Pull request merged**